sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -70,9 +70,9 @@ class TemplateUserOptions(PipelineOptions):
       @classmethod
 
       def _add_argparse_args(cls, parser):
-        parser.add_value_provider_argument('--vp-arg1', default='start')
-        parser.add_value_provider_argument('--vp-arg2')
-        parser.add_argument('--non-vp-arg')
+        parser.add_value_provider_argument('--vp_arg1', default='start')
```

Comment:
Why do we change these examples? Are not -vp-arg1 a valid argument?

Reply:
I want to discourage using dashes in the name. Argparse considers them valid, but as @udim has checked, flags with dashes passed in `**kwargs` parameter of PipelineOptions constructor will not overwrite flags with dashes defined in `_add_argparse_args()` method. Argparse normalizes dashes to underscores, but Beam does not. I was thinking about sending a PR to emit a warning when dashes are used to encourage users to stop using them.

Reply:
A warning would be good in that case. (We can resolve this comment.)